social_core/backends/mediawiki.py

```diff
@@ -0,0 +1,175 @@
+                       '?title=Special:Oauth/token'
+    IDENTIFY_URL = 'https://en.wikipedia.org/w/index.php' + \
+                   '?title=Special:OAuth/identify'
+
+    def unauthorized_token(self):
+        """
+        Return request for unauthorized token (first stage)
+
+        Mediawiki request token is requested from e.g.:
+         * https://en.wikipedia.org/w/index.php?title=Special:OAuth/initiate
+         * 'callback_uri' needs to be 'oob'
+        """
+        params = self.request_token_extra_arguments()
+        params.update(self.get_scope_argument())
+        key, secret = self.get_key_and_secret()
+        decoding = None if six.PY3 else 'utf-8'
+
+        response = self.request(
+            self.setting('REQUEST_TOKEN_URL'),
+            params=params,
+            auth=OAuth1(key, secret, callback_uri='oob',
+                        decoding=decoding),
+            method=self.REQUEST_TOKEN_METHOD
+        )
+
+        content = response.content.decode()
+
+        return content
+
+    def oauth_authorization_request(self, token):
+        """
+        Generates the URL for the authorization link
+
+        For English Wikipedia the URL is either:
+         * https://en.wikipedia.org/w/index.php
+                ?title=Special:Oauth/authorize'
+         * https://en.wikipedia.org/w/index.php
+                ?title=Special:Oauth/authenticate'
+        """
+        if not isinstance(token, dict):
+            token = parse_qs(token)
+
+        params = {}
+
+        oauth_token = token.get(self.OAUTH_TOKEN_PARAMETER_NAME)[0]
+        params[self.OAUTH_TOKEN_PARAMETER_NAME] = oauth_token
+
+        state = self.get_or_create_state()
+        params[self.REDIRECT_URI_PARAMETER_NAME] = self.get_redirect_uri(state)
+
+        base_url, oauth_title_param = self.setting('AUTHORIZATION_URL').split('?')
+        _, oauth_title = oauth_title_param.split('=')
+
+        params['title'] = oauth_title
+
+        return '{0}?{1}'.format(base_url, urlencode(params))
+
+    def access_token(self, token):
+        """
+        Fetches the Mediawiki access token.
+        """
+        url = self.access_token_url()
+        url_base, title = url.split('?')
+        _, title_param = title.split('=')
+        auth_token = self.oauth_auth(token)
+
+        response = requests.post(url=url_base,
+                                 params={'title': title_param},
+                                 auth=auth_token)
+
+        credentials = parse_qs(response.content)
+        oauth_token_key = credentials.get(b('oauth_token'))[0]
+        oauth_token_secret = credentials.get(b('oauth_token_secret'))[0]
+        oauth_token_key = oauth_token_key.decode()
+        oauth_token_secret = oauth_token_secret.decode()
+
+        access_token = {'oauth_token': oauth_token_key,
+                        'oauth_token_secret': oauth_token_secret}
+        return access_token
+
+    def get_user_details(self, token):
```

Comment:
The signature of `BaseAuth::get_user_details` is `(self, response)`, not `(self, token)`.